cirq/ops/controlled_operation.py

```diff
@@ -119,8 +119,22 @@ def _resolve_parameters_(self, resolver):
         new_sub_op = protocols.resolve_parameters(self.sub_operation, resolver)
         return ControlledOperation(self.controls, new_sub_op)
 
-    def _trace_distance_bound_(self) -> float:
-        return protocols.trace_distance_bound(self.sub_operation)
+    def _trace_distance_bound_(self):
+        if self._is_parameterized_():
+            return 1
+        if protocols.has_unitary(self.sub_operation):
+            angles = np.sort(
+                np.append(
+                    np.angle(
+                        np.linalg.eigvals(protocols.unitary(
+                            self.sub_operation))), 0))
```

Comment:
It's too bad that this needs the unitary even if sub_operation implementes `_trace_distance_bound_`.

Reply:
It technically only needs the unitary of the sub_operation, so it isn't as expensive as it could have been. It is unfortunate though.